core/src/main/scala/kafka/coordinator/group/CoordinatorLoaderImpl.scala

```diff
@@ -156,17 +164,42 @@ class CoordinatorLoaderImpl[T](
               } else {
                 batch.asScala.foreach { record =>
                   numRecords = numRecords + 1
```

Comment:
(this comment is unrelated to this PR)

the `numBytes` is based on whole `memoryRecords`. By contrast, the `numRecords` does not include control batch. Is it expected?

Reply:
Yeah, the goal was really to count the records. I think that we could also count the control records but with a separate counter. I will do this as a follow up.